valtech-sd/amqp-cacoon
# PPAA-423 Update to receive other loggers

##  [Update AMQP Caccon to receive other loggers](https://jira.valtech.com/browse/PPAA-423)

### Changes
- [x] added ILogger interface
- [x] updated tests (removed log4js) 
- [x] removed log4js from package.json
- [x] updated version in package.json
- [x] updated package-lock.json using node version 16.7
 

### Not changed
- passing a logger was already optional
- the example keeps using log4js

### How to test

1. npm run test
2. run the example
3. You should be able to instantiate the `AmqpCaccon` class passing different loggers.
